src/mdlComponentHandler.js

```diff
@@ -389,7 +374,15 @@ componentHandler = (function() {
      * @param  {!Node} node the node to be downgraded
      */
     var downgradeNode = function(node) {
-      deconstructComponentInternal(findCreatedComponentByNodeInternal(node));
+      var found = [];
+      createdComponents_.forEach(function(item) {
+        if (item.element_ === node) {
+          found.push(item);
```

Comment:
Testing this again now. When I attempted it before I had some buggy behavior downgrading buttons with ripples. Of course, it also could have been my previous implementation being buggy so testing again to verify.